### PR TITLE
atuin crypto blog post

### PR DESCRIPTION
**Pull request by conradludgate**

### CODE REVIEW

File: docs/blog/2023/07/07-new-encryption/index.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+title: Atuin's new encryption scheme
+description: Details about the new encryption system for Atuin
+slug: new-encryption
+authors: [conrad]
+tags: [insights]
+---
+
+End-to-end encryption is an essential component of Atuin.
+One of our core philosophies, when we created the sync service, is that
+we want no reason to worry about user data. The shell is a very
```
Comment (eopb) [inline docs/blog/2023/07/07-new-encryption/index.md:11]:
```suggestion
One of our core philosophies, when we created the sync service, was that
we didn't want to worry about storing user data. The shell is a very
```

File: docs/blog/2023/07/07-new-encryption/index.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+their local encryption scheme is designed such that encrypted data is safe to be shared publically.
+Their V4 scheme also uses the XChaCha20 cipher which I was initially planning to use.
+
+In the end, I bit the bullet and decided to use the standard.
+The nice thing with secretbox is that existing implementations in other languages are widely available.
+Making it easy to implement sync in third parties. If we implemented our own scheme,
+that would make it much easier for third parties to make mistakes if they wanted to use the sync data directly.
+
+Using PASETO, there are existing implementations that we didn't have to write.
+This means that we don't build software doomed to die a lonely death. It also means
```
Comment (eopb) [inline docs/blog/2023/07/07-new-encryption/index.md:130]: :rofl:

File: docs/blog/2023/07/07-new-encryption/index.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+However, there is still the issue of key leaking. We have no key-upgrade policy.
+If a key is leaked, maybe through a side-channel attack, a social attack, or malware,
+then the only solution is to create a new account with a new key.
+
+This is partially an issue.
+
+## Change
+
+While researching these systems, I have learnt a lot more new techniques that modern cryptographic systems use. While the analysis above indicates that we are protected, there might be attacks we are unaware of, so keeping up with modern research is important.
```
Comment (eopb) [inline docs/blog/2023/07/07-new-encryption/index.md:89]:
```suggestion
While researching these systems, I learnt of many new cryptographic techniques that some modern systems use. While the analysis above indicates that we are protected, there might be attacks we are unaware of, so keeping up with modern research is important.
```

File: docs/blog/2023/07/07-new-encryption/index.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+The idea here is that each payload has an ephemeral encryption key.
+This is then encrypted (wrapped) using the master key and stored with the data.
+
+Initially, this seemed less secure to me. However, my analysis seems to point out that
+the master key is less vulnerable to side-channel attacks since it is less used.
+It also offers no decrease in security since brute-forcing the master key from an
+ephemeral data key is just as hard as it is for any message. In the end, it's like
+a password manager for your encrypted data.
+
+This unlocks some potential future upgrades.
```
Comment (eopb) [inline docs/blog/2023/07/07-new-encryption/index.md:105]:
```suggestion
This would unlock some potential future upgrades.
```

**Review by eopb — COMMENTED** [review]
Great blog post Conrad. I've added some suggestions for minor things I felt could be tweaked. Take from it any ones you find helpful.

**Review by ellie — COMMENTED** [review]
Fantastic post, thanks! I'm going to give it another read through in the morning (probs once the feedback has been merged or not merged) but generally looks good to go 🚀

**Comment by conradludgate** [comment]
Thanks @eopb 🙏

**Review by ellie — APPROVED** [review]
nice one! 🚀

**Pull request merged**